lib/form/js/repeatable.js

```diff
@@ -0,0 +1,245 @@
+              if (!isNaN(index)) {
+                splitChild[splitChild.length-1] = index - 1;
+                var oldChildId = child.id;
+                child.id = splitChild.join('--');
+
+                // Maintains state of allInputs
+                if (allInputs.hasOwnProperty(oldChildId)) {
+                  allInputs[child.id] = allInputs[oldChildId];
+                  delete allInputs[oldChildId];
+                }
+                if (allIDs.hasOwnProperty(oldChildId)) {
+                  // Maintains state of allIDs for changing inputs
+                  allIDs[child.id] = allIDs[oldChildId];
+                  delete allIDs[oldChildId];
+                }
+              }
+              var childName = child.getAttribute('name');
+              if (childName) {
+                childName = childName.split('--');
+                index = childName[childName.length-1];
+                if (!isNaN(index)) {
+                  childName[childName.length-1] = index - 1;
+                  child.setAttribute('name', childName.join('--'));
+                }
+              }
+            }
+            else if (child.htmlFor && child.htmlFor != "") {
+              splitChild = child.htmlFor.split('--');
+              htmlFor = splitChild[splitChild.length-1];
+              if (!isNaN(htmlFor)) {
+                splitChild[splitChild.length-1] = htmlFor - 1;
+                child.htmlFor = splitChild.join('--');
+              }
+            }
+          });
+        }
+        siblingNode = siblingNode.nextElementSibling;
+      }
+
+      if (allRepeatables[input].length == 1) {
+        var container, nodeId, id, name, htmlFor;
+        container = parent.querySelectorAll('.form-repeatable')[0];
+        nodeId = splitPop(container.id, '--')
+        container.id = nodeId;
+        childNodes = container.querySelectorAll('*');
+        childNodes.forEach(function(child) {
+          if (child.id != "") {
+            child.id = splitPop(child.id, '--');
+            name = child.getAttribute('name');
+            if(name) {
+              name = splitPop(name, '--');
+              child.setAttribute('name', name);
+            }
+          }
+          else if (child.htmlFor && child.htmlFor != "") {
+            child.htmlFor = splitPop(child.htmlFor, '--');
+          }
+        });
+        container.removeChild(container.querySelectorAll('#' + input + '-del')[0]);
+      }
+    };
+
+    // Handles add button event
+    var addHandler = function (e) {
+
+      // Holds the first input element of instance
+      var focusChild;
+
+      // selects first container of instance
+      var container = e.currentTarget.parentNode.querySelectorAll('div.form-repeatable')[0];
+
+      // Modify name of initial instance and adds delete button
+      if (allRepeatables[input].length == 1) {
+        var newElement = document.createElement('button');
+        newElement.type = 'button';
+        newElement.innerHTML = 'Delete';
```

Comment:
Please do not use `innerHTML`, it's creates the potential for security vulnerabilities. Please use `textContent` instead